src/main/java/frc/robot/subsystems/intake/Intake.kt

```diff
@@ -50,6 +53,14 @@ abstract class Intake : Subsystem {
     }
 
     fun moveIntake(position: Rotation2d, velocity: Rotation2d) {
+        stallDetector.calculate(inputs.armVelocity.radians, position.radians)
+
+        // only reset if we're moving towards the hard stop and not making progress
+        if (stallDetector.i > 0.5 && position.radians - inputs.position.radians > 0.0) {
```

Comment:
I think an approach with getAppliedOutput() might be a bit cleaner. My main concerns with using the integral of a pid is that as soon as it crosses the no no threshold, It will never decrease, unless you only consider a certain time frame, but even then there would be some error. Instead, I would get the derivative and compare that with the applied output.

Reply:
Applied output returns the percent output of the motor. When multiplied by the max voltage, It will give us the voltage being applied the motor. If the absolute value of the derivative of the motor position (speed) is close to zero, and the applied output is high, we know we have hit a hard stop.

Reply:
Okay, that sounds like it would work. But, if the desired motor position has changed, that will increase the applied output without immediately affecting speed. Wouldn't this cause a misfire of this check?

Reply:
I believe it will affect speed immediately, but even if it did we could probably use the trapezoidal motion profile to account for that.